File: OpenCV/10_segmentacion/cpp/main.cpp
#include <opencv2/opencv.hpp>
int main(){ auto g=cv::imread("input.jpg",cv::IMREAD_GRAYSCALE); if(g.empty()) return 1; cv::Mat o,a; cv::threshold(g,o,0,255,cv::THRESH_BINARY|cv::THRESH_OTSU); cv::adaptiveThreshold(g,a,255,cv::ADAPTIVE_THRESH_GAUSSIAN_C,cv::THRESH_BINARY,31,5); cv::imwrite("output_otsu.png",o); cv::imwrite("output_adaptive.png",a); }

File: OpenCV/10_segmentacion/python/main.py
import cv2
g=cv2.imread("input.jpg",cv2.IMREAD_GRAYSCALE); assert g is not None
_,otsu=cv2.threshold(g,0,255,cv2.THRESH_BINARY+cv2.THRESH_OTSU); adap=cv2.adaptiveThreshold(g,255,cv2.ADAPTIVE_THRESH_GAUSSIAN_C,cv2.THRESH_BINARY,31,5); cv2.imwrite("output_otsu.png",otsu); cv2.imwrite("output_adaptive.png",adap)
